lib/pheromone/publishable.rb

```diff
@@ -94,8 +94,9 @@ def send_message(options, current_event)
 
       def message_meta_data(options, current_event)
         metadata = options[:metadata]
-        return { event: current_event, entity: self.class.name } if metadata.blank?
-        return metadata if metadata.is_a?(Hash)
+        default_metadata = { event: current_event, entity: self.class.name }
+        return default_metadata if metadata.blank?
+        return default_metadata.merge(metadata) if metadata.is_a?(Hash)
         call_proc_or_instance_method(metadata)
```

Comment:
we need to also take this into account. so need to merge default metdata to the proc's result, and the result should also be a Hash ideally